sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientTests.cs

```diff
@@ -1326,6 +1327,244 @@ public async Task OpenReadAsync_InvalidParameterTests()
+                () => outputStream.Seek(-10, SeekOrigin.Begin),
+                new ArgumentException("New offset cannot be less than 0.  Value was -10"));
+        }
+
+        [Test]
+        public async Task OpenReadAsync_Seek_NewPositionGreaterThanBlobLength_AllowBlobModificationsFalse()
+        {
+            int size = Constants.KB;
+            await using DisposingContainer test = await GetTestContainerAsync();
+
+            // Arrange
+            var data = GetRandomBuffer(size);
+            BlockBlobClient blob = InstrumentClient(test.Container.GetBlockBlobClient(GetNewBlobName()));
+            using Stream stream = new MemoryStream(data);
+            await blob.UploadAsync(stream);
+
+            // Act
+            Stream outputStream = await blob.OpenReadAsync().ConfigureAwait(false);
+            TestHelper.AssertExpectedException<ArgumentException>(
+                () => outputStream.Seek(1025, SeekOrigin.Begin),
+                new ArgumentException("You cannot seek past the end of a stream created by OpenReadStream that does not allow modifications to the underlying blob or file."));
+        }
+
+        [Test]
+        public async Task OpenReadAsync_Seek_NewPositionGreaterThanBlobLength_AllowBlobModificationsTrue()
+        {
+            int size = Constants.KB;
+            await using DisposingContainer test = await GetTestContainerAsync();
+
+            // Arrange
+            var data = GetRandomBuffer(size);
+            BlockBlobClient blob = InstrumentClient(test.Container.GetBlockBlobClient(GetNewBlobName()));
+            using Stream stream = new MemoryStream(data);
+            await blob.UploadAsync(stream);
+
+            BlobOpenReadOptions options = new BlobOpenReadOptions(allowModifications: true);
+
+            // Act
+            Stream outputStream = await blob.OpenReadAsync(options: options).ConfigureAwait(false);
+            outputStream.Seek(size + 10, SeekOrigin.Begin);
+            Assert.AreEqual(size + 10, outputStream.Position);
+        }
+
+        [Test]
+        public async Task OpenReadAsync_Seek_NewPositionGreaterThanBlobLength_AllowBlobModificationsTrue_NoData()
+        {
+            int size = Constants.KB;
+            await using DisposingContainer test = await GetTestContainerAsync();
+
+            // Arrange
+            var data = GetRandomBuffer(size);
+            BlockBlobClient blob = InstrumentClient(test.Container.GetBlockBlobClient(GetNewBlobName()));
+            using Stream stream = new MemoryStream(data);
+            await blob.UploadAsync(stream);
+
+            BlobOpenReadOptions options = new BlobOpenReadOptions(allowModifications: true);
+
+            // Act
+            Stream outputStream = await blob.OpenReadAsync(options: options).ConfigureAwait(false);
+            outputStream.Seek(size + 10, SeekOrigin.Begin);
+            byte[] outputArray = new byte[1];
+
+            // Assert
+            Assert.Zero(stream.Read(outputArray, 0, 1));
```

Comment:
We should try to read from stream returned from OpenRead instead of original memory stream. When I try that it throws 416.
```suggestion
            Assert.Zero(outputStream .Read(outputArray, 0, 1));
```

Reply:
You're right.  How do you think we should handle this case?  We could catch the exception and return 0, maybe there is a better option.

Reply:
I think @tg-msft  suggested to return 0. 
One thing to check: what happens if we seek almost till the real end and provide buffer that would result in range spaning across the end, i.e.:
- create blob with length 1024
- open read and seek till 1100
- append 100 bytes to blob
- read providing buffer that has 100 bytes+